src/varname.jl

```diff
@@ -418,28 +418,34 @@ _issubrange(i::ConcreteIndex, j::Colon) = true
 # we preserve the status quo, but I'm confused.
 _issubrange(i::Colon, j::ConcreteIndex) = true
 
+"""Turn any `AbstractRange` into an equivalent `UnitRange` or `StepRange`."""
+_normalizerange(r::AbstractRange) = (:)(r...)
+_normalizerange(x) = x
+
 """
     concretize(l::Lens, x)
 
 Return `l` instantiated on `x`, i.e. any runtime information evaluated using `x`.
 """
 concretize(I::Lens, x) = I
-concretize(I::DynamicIndexLens, x) = IndexLens(I.f(x))
+concretize(I::IndexLens, x) = IndexLens(_normalizerange.(Base.to_indices(x, I.indices)))
+concretize(I::DynamicIndexLens, x) = IndexLens(_normalizerange.(I.f(x)))
```

Comment:
No need for broadcasting here:
```suggestion
concretize(I::IndexLens, x) = IndexLens(map(_normalizerange, Base.to_indices(x, I.indices)))
concretize(I::DynamicIndexLens, x) = IndexLens(map(_normalizerange, I.f(x)))
```